code/core/src/main/java/com/adobe/marketing/mobile/services/ui/vnext/message/InAppMessageSettings.kt

```diff
@@ -0,0 +1,234 @@
+
+        /**
+         * Sets the vertical alignment of the message on the screen.
+         * @param verticalAlignment the vertical [MessageAlignment] of the message on the screen
+         */
+        fun verticalAlignment(verticalAlignment: MessageAlignment) =
+            apply { this.verticalAlignment = verticalAlignment }
+
+        /**
+         * Sets the horizontal alignment of the message on the screen.
+         * @param horizontalAlignment the horizontal [MessageAlignment] of the message on the screen
+         */
+        fun horizontalAlignment(horizontalAlignment: MessageAlignment) =
+            apply { this.horizontalAlignment = horizontalAlignment }
+
+        /**
+         * Sets the animation to use when displaying the message.
+         * @param displayAnimation the [MessageAnimation] to use when displaying the message
+         */
+        fun displayAnimation(displayAnimation: MessageAnimation) =
+            apply { this.displayAnimation = displayAnimation }
+
+        /**
+         * Sets the animation to use when dismissing the message.
+         * @param dismissAnimation the [MessageAnimation] to use when dismissing the message
+         */
+        fun dismissAnimation(dismissAnimation: MessageAnimation) =
+            apply { this.dismissAnimation = dismissAnimation }
+
+        /**
+         * Sets the color of the backdrop behind the message. This is the color behind the message when the message is taking over the UI.
+         * @param backgroundColor the hex color of the backdrop behind the message.
+         */
+        fun backgroundColor(backgroundColor: String) =
+            apply { this.backgroundColor = backgroundColor }
+
+        /**
+         * Sets the opacity of the backdrop behind the message. This is the opacity behind the message when the message is taking over the UI.
+         * @param backdropOpacity the opacity of the backdrop behind the message. This should be a value between 0.0f (transparent) and 1.0f(opaque).
+         */
+        fun backdropOpacity(backdropOpacity: Float) =
+            apply { this.backdropOpacity = backdropOpacity }
+
+        /**
+         * Sets the corner radius of the message.
+         * @param cornerRadius the corner radius of the message
+         */
+        fun cornerRadius(cornerRadius: Float) = apply { this.cornerRadius = cornerRadius }
+
+        /**
+         * Configures whether the message should take over the UI.
+         * @param shouldTakeOverUi whether the message should take over the UI. Should be set to true if the message should take over the UI, false otherwise.
+         */
+        fun shouldTakeOverUi(shouldTakeOverUi: Boolean) =
+            apply { this.shouldTakeOverUi = shouldTakeOverUi }
+
+        /**
+         * Sets the asset map for the message. This is a map of asset names to asset URLs.
+         * @param assetMap the asset map for the message
+         */
+        fun assetMap(assetMap: MutableMap<String, String>) =
```

Comment:
Any reason this needs to be MutableMap<String, String> instead of Map<String, String>?

Reply:
Good catch. This should have been immutable. Will change it.